Modules/posixmodule.c

```diff
@@ -322,29 +324,35 @@ corresponding Unix manual entries for more information on calls.");
 #  if defined(TIOCGWINSZ)
 #    define TERMSIZE_USE_IOCTL
 #  endif
-#endif /* MS_WINDOWS */
+#endif /* MS_WINDOWS && !MS_WINDOWS_NON_DESKTOP */
 
 /* Various compilers have only certain posix functions */
 /* XXX Gosh I wish these were all moved into pyconfig.h */
 #if defined(__WATCOMC__) && !defined(__QNX__)           /* Watcom compiler */
 #  define HAVE_OPENDIR    1
 #  define HAVE_SYSTEM     1
 #  include <process.h>
-#else
-#  ifdef _MSC_VER
-     /* Microsoft compiler */
+#elif defined( _MSC_VER)
+  /* Microsoft compiler */
+#  ifndef MS_WINDOWS_NON_DESKTOP
 #    define HAVE_GETPPID    1
 #    define HAVE_GETLOGIN   1
 #    define HAVE_SPAWNV     1
 #    define HAVE_EXECV      1
 #    define HAVE_WSPAWNV    1
 #    define HAVE_WEXECV     1
-#    define HAVE_PIPE       1
 #    define HAVE_SYSTEM     1
 #    define HAVE_CWAIT      1
-#    define HAVE_FSYNC      1
-#    define fsync _commit
-#  endif  /* _MSC_VER */
+#  endif /* !MS_WINDOWS_NON_DESKTOP */
+#  define HAVE_PIPE       1
+#  define HAVE_FSYNC      1
+#  define fsync _commit
+#  ifdef MS_WINDOWS_GAMES
+#    define dup2 _dup2
+#    define umask _umask
+#    define close _close
+#    define isatty _isatty
```

Comment:
As [discussed](https://learn.microsoft.com/en-us/cpp/c-runtime-library/compatibility?view=msvc-170) in the documentation, define `_CRT_DECLARE_NONSTDC_NAMES` in "PC/pyconfig.h" to force declaring the POSIX names.

> To suppress the warning, define `_CRT_NONSTDC_NO_WARNINGS` before including any headers in code that uses the original POSIX names. Because the C standard doesn't allow non-standard names in header files, by default [/std:c11](https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-170) and [/std:c17](https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-170) don't expose the default names for POSIX functions, types, and macros. If these names are necessary, define `_CRT_DECLARE_NONSTDC_NAMES` to expose them.

For example:

```c
#ifndef _CRT_DECLARE_NONSTDC_NAMES
#define _CRT_DECLARE_NONSTDC_NAMES 1
#endif
#ifndef _CRT_NONSTDC_NO_WARNINGS
#define _CRT_NONSTDC_NO_WARNINGS 1
#endif
```

We've been defining `_CRT_NONSTDC_NO_DEPRECATE`. This in turn causes `_CRT_NONSTDC_NO_WARNINGS` to be defined, but only the latter is documented nowadays.

Reply:
This does not help here, since on the xbox the deprecated posix variants are not implemented. However the underscore variants do exist.

Reply:
Yes, but this leads to linker errors about missing symbols like `__imp_isatty`. A developer question from 2014 in the xbox forums suggests, that it is required to use the underscore version on the xbox. However I will ask whether thats actually the case. Could be I just need to link another lib as well.

Reply:
Apparently it's "oldnames.lib". See [backward compatibility](https://learn.microsoft.com/en-us/cpp/c-runtime-library/backward-compatibility?view=msvc-170).

> For compatibility between product versions, the library OLDNAMES.LIB maps old names to new names. For instance, `open` maps to `_open`. You must explicitly link with OLDNAMES.LIB only when you compile with the following combinations of command-line options:
>
> * `/Zl` (omit default library name from object file) and `/Ze` (the default: use Microsoft extensions)
>
> * `/link` (linker-control), `/NOD` (no default-library search), and `/Ze`

Reply:
thanks a lot :)
I was sorely searching for this.